roles/kubernetes/preinstall/tasks/set_resolv_facts.yml

```diff
@@ -50,10 +50,14 @@
     dhclientconffile: /etc/dhclient.conf
   when: dhclient_stat.stat.exists
 
+- name: check if /etc/dhcp/dhclient.conf exists
+  stat: path=/etc/dhcp/dhclient.conf
+  register: dhcp_dhclient_stat
+
 - name: target dhclient conf file for /etc/dhcp/dhclient.conf
   set_fact:
     dhclientconffile: /etc/dhcp/dhclient.conf
-  when: not dhclient_stat.stat.exists
+  when: dhcp_dhclient_stat.stat.exists
```

Comment:
How about we consolidate this using with_first_found?

```- name: target dhclient conf file
  set_fact:
    dhclientconffile: "{{item}}"
  with_first_found:
    - files:
      - dhclient.conf
    - paths:
      - /etc/dhcp
      - /etc
     skip:true
```
Your logic above in the other 2 logs can be unchanged.